### PR TITLE
CG cards modal window feature

### PR DESCRIPTION
**Pull request by AvocadoCoding**
HTML, JS and CSS modified to add buttons to cards that open and close a modal window

### CODE REVIEW

**Comment by Nwinch1512** [comment]
Hi Chris, thanks for doing this - it looks great and is working really well!!  I've just noticed one minor thing - there's a semi-colon missing on the end of line 107 in the script file.  Also another minor thing - for some reason the size of the recipe cards has altered slightly and the image is starting the repeat at the bottom of the recipe div....

Hi Naomi, yes I've added the semi-colan and made the other changes you've identified. With the cards I think the addition of the button has made the cards bigger, and the image is now repeating to fill the extra space. I think we can leave this as it is for the moment and then come back to the structure and styling of the cards later?

File: assets/script.js [hunk 1]
```diff
@@ -104,7 +104,7 @@ function displayRecipes(url) {
       let recipeDiv = $("<div>")
         .addClass("card-body col-lg-3 col-md-3 col-sm-6 text-center")
         // .css("background-color", "rgb(107,101,75)");
-        .css("background-image", `url(${mealImgURL})`);
+        .css("background-image", `url(${mealImgURL})`)
```
Comment (Nwinch1512) [inline assets/script.js]: Here's where I think it's missing a semi-colon

File: assets/script.js [hunk 2]
```diff
@@ -123,10 +123,55 @@ function displayRecipes(url) {
...
+
+
+
+// Code to control opening and closing of modal
+
+// Get the modal div from HTML
+var modal = $("#myModal");
+
+// Get the button that opens the modal
+var modalBtn = $(".modal-btn");
```
Comment (Nwinch1512) [inline assets/script.js]: I don't think this is used anywhere? The cooking instructions buttons won't exist at this stage

File: assets/script.js [hunk 3]
```diff
@@ -123,10 +123,55 @@ function displayRecipes(url) {
...
+var span = $(".close")[0];
+
+// When the user clicks the button, open the modal. This function is used
+// so appended buttons added by JS to document can be clicked on
+
+$(document).on("click",".modal-btns",openModal);
+
+function openModal (event){
+  event.preventDefault();
+  console.log("click test 1");
```
Comment (Nwinch1512) [inline assets/script.js]: I've tested this locally and it works great! Could remove the console log test now we know this handler is used?

**Review by Nwinch1512 — COMMENTED** [review]
Testing out the proper code review process by adding comments in the code lines hence the additional review.

This is fantastic! I thought this would take much longer. A few optional comments to address then we can get it merged.

File: assets/script.js (hunk 1 above)
Comment (AvocadoCoding) [reply assets/script.js]: Hi Naomi, yes agree there is a semi-colan missing there, I'll add that

File: assets/script.js (hunk 2 above)
Comment (AvocadoCoding) [reply assets/script.js]: Yes you are write, I put it in initially before switching to the $(document).on("click",".modal-btns",openModal) when I realised it wouldn't work, but forgot to take those lines out. I'll remove lines 149 and 150.

File: assets/script.js (hunk 3 above)
Comment (AvocadoCoding) [reply assets/script.js]: Ok yes, will remove.

**Review by Nwinch1512 — APPROVED** [review]
Thanks Chris - this is looking great!

**Pull request merged**